schema/spacewalk/postgres/procs/lookup_transaction_package.sql

```diff
@@ -40,13 +41,16 @@ begin
     end if;
 
     n_id := lookup_package_name(n_in);
-    e_id := lookup_evr(e_in, v_in, r_in);
     p_arch_id := null;
 
     if a_in is not null then
         p_arch_id := lookup_package_arch(a_in);
+        SELECT t.label into type from rhnpackagearch pa join rhnarchtype t
+         on t.id = pa.arch_type_id where pa.id = p_arch_id;
     end if;
 
+    e_id := lookup_evr(e_in, v_in, r_in, type);
```

Comment:
When a_in is null, this will create a EVR without type.
This function is used in backend with an explicit arch set to None.

https://github.com/uyuni-project/uyuni/blob/debvercmp-uyuni/backend/server/rhnServer/server_kickstart.py#L172-L207